serve/source_v0_test.go

```diff
@@ -56,72 +56,81 @@ func TestSourceSuccessV1(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to dial bufnet: %v", err)
 	}
-	c, err := clients.NewClient(ctx, specs.RegistryGrpc, "", "", clients.WithGRPCConnection(conn), clients.WithNoSentry())
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer func() {
-		if err := c.Terminate(); err != nil {
-			t.Fatal(err)
-		}
-	}()
+	c := pb.NewSourceClient(conn)
 
-	name, err := c.Name(ctx)
+	getNameResponse, err := c.GetName(ctx, &pb.GetName_Request{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if name != "testPlugin" {
-		t.Fatalf("expected name to be testPlugin but got %s", name)
+	if getNameResponse.Name != "testPlugin" {
+		t.Fatalf("expected name to be testPlugin but got %s", getNameResponse.Name)
 	}
 
-	version, err := c.Version(ctx)
+	getVersionRes, err := c.GetVersion(ctx, &pb.GetVersion_Request{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if version != "v1.0.0" {
-		t.Fatalf("Expected version to be v1.0.0 but got %s", version)
+	if getVersionRes.Version != "v1.0.0" {
+		t.Fatalf("Expected version to be v1.0.0 but got %s", getVersionRes.Version)
 	}
 
-	tables, err := c.GetTables(ctx)
+	getTablesRes, err := c.GetTables(ctx, &pb.GetTables_Request{})
 	if err != nil {
 		t.Fatal(err)
 	}
+	var tables schema.Tables
+	if err := json.Unmarshal(getTablesRes.Tables, &tables); err != nil {
+		t.Fatal(err)
+	}
 	if len(tables) != 2 {
 		t.Fatalf("Expected 2 tables but got %d", len(tables))
 	}
-
-	if err := c.Init(ctx, specs.Source{
+	spec := specs.Source{
 		Name:         "testSourcePlugin",
 		Version:      "v1.0.0",
 		Path:         "cloudquery/testSourcePlugin",
 		Registry:     specs.RegistryGithub,
 		Tables:       []string{"test_table"},
 		Spec:         TestSourcePluginSpec{Accounts: []string{"cloudquery/plugin-sdk"}},
 		Destinations: []string{"test"},
-	}); err != nil {
+	}
+	specBytes, err := json.Marshal(spec)
+	if err != nil {
 		t.Fatal(err)
 	}
-
-	resources := make(chan []byte, 2)
-	if err := c.Sync(ctx,
-		resources); err != nil {
+	if _, err := c.Init(ctx, &pb.Init_Request{Spec: specBytes}); err != nil {
 		t.Fatal(err)
 	}
-	close(resources)
 
-	totalResources := 0
-	for resourceB := range resources {
+	syncClient, err := c.Sync(ctx, &pb.Sync_Request{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	var resources []schema.DestinationResource
+	for {
+		r, err := syncClient.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
 		var resource schema.DestinationResource
-		if err := json.Unmarshal(resourceB, &resource); err != nil {
-			t.Fatalf("failed to unmarshal resource: %v", err)
+		if err := json.Unmarshal(r.Resource, &resource); err != nil {
+			t.Fatal(err)
```

Comment:
```suggestion
			t.Fatal("failed to unmarshal resource: %v", err)
```

Reply:
because it's just test then I prefer to to add formatting because it printing the line anyway.

Reply:
some places had formatting and some didn't so I tried to make it consistent without formatting in test and just call `t.Fatal`